frontend/src/static/js/v_authorship.js

```diff
@@ -41,23 +41,66 @@ window.vAuthorship = {
       totalBlankLineCount: '',
       filesSortType: 'lineOfCode',
       toReverseSortFiles: true,
-      filterSearch: '*',
+      searchBarValue: '',
     };
   },
 
   watch: {
-    filterType() {
-      if (this.filterType === 'checkboxes') {
-        const searchBar = document.getElementById('search');
-        searchBar.value = '';
-        this.filterSearch = '*';
-      } else {
-        this.selectedFileTypes = this.fileTypes.slice();
+    filesSortType() {
+      window.addHash('authorshipSortBy', this.filesSortType);
+      window.encodeHash();
+    },
+
+    toReverseSortFiles() {
+      window.addHash('reverseAuthorshipOrder', this.toReverseSortFiles);
+      window.encodeHash();
+    },
+
+    isLoaded() {
+      if (this.isLoaded) {
+        this.retrieveHashes();
       }
     },
   },
 
   methods: {
+    retrieveHashes() {
+      const {
+        addHash,
+        decodeHash,
+        encodeHash,
+        hashParams: hash,
+      } = window;
+
+      decodeHash();
+
+      switch (hash.authorshipSortBy) {
+      case 'path':
+      case 'fileName':
+      case 'fileType':
+        this.filesSortType = hash.authorshipSortBy;
+        break;
+      default:
+        // Invalid value, use the default value of 'lineOfCode'
+      }
+
+      this.toReverseSortFiles = hash.reverseAuthorshipOrder !== 'false';
+
+      if ('authorshipFilesGlob' in hash) {
+        this.indicateSearchBar();
+        this.searchBarValue = hash.authorshipFilesGlob;
+      } else if ('authorshipFileTypes' in hash) {
+        const parsedFileTypes = hash.authorshipFileTypes.split('~');
```

Comment:
Do you think we can define a global constant for delimiters? In the future, when we need to restore the checked file types in v_summary etc, it may be useful.